### PR TITLE
Feature/KITOSUDV-3552 (scheduled inspection date)

### PR DESCRIPTION
**Pull request by a-naskret**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [x] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [x] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Core.DomainModel/GDPR/DataProcessingRegistration.cs [hunk 1]
```diff
@@ -17,7 +17,6 @@ namespace Core.DomainModel.GDPR
     public class DataProcessingRegistration :
         HasRightsEntity<DataProcessingRegistration, DataProcessingRegistrationRight, DataProcessingRegistrationRole>,
         IHasName,
-        IOwnedByOrganization,
```
Comment (a-naskret) [inline Core.DomainModel/GDPR/DataProcessingRegistration.cs:20]: "IEntityWithAdvices" also inherits "IOwnedByOrganization"

File: Presentation.Web/app/components/data-processing/tabs/data-processing-registration-tab-oversight.view.html [hunk 2]
```diff
@@ -65,6 +65,16 @@ <h4 data-element-type="dpaMainDetailHeader">{{vm.headerName}}</h4>
                                       ng-disabled="!vm.hasWriteAccess" />
                     </div>
                 </div>
+
+                <div class="form-group">
+                    <!-- Date of scheduled inspection -->
+                    <div class="form-group">
+                        <label for="scheduledInspection" class="control-label small">Scheduled inspection date</label>
```
[the remark this answers is not in the document]
Comment (mrjsawdk) [reply Presentation.Web/app/components/data-processing/tabs/data-processing-registration-tab-oversight.view.html:72]: Deleting the date does not work... it retains the value

File: Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationWriteServiceTest.cs [hunk 3]
```diff
@@ -1365,6 +1365,67 @@ public void Can_Create_With_OversightData_OversightIntervalRemark_Set_To_NoChang
             _dprServiceMock.Verify(x => x.UpdateOversightIntervalRemark(It.IsAny<int>(), It.IsAny<string>()), Times.Never);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Can_Create_With_OversightScheduledInspectionDate_OversightIntervalRemark(bool inputIsNull)
```
Comment (mrjsawdk) [inline Tests.Unit.Core.ApplicationServices/ApplicationServices/GDPR/DataProcessingRegistrationWriteServiceTest.cs:1371]: why us OversightIntervalRemark part of the names of your test methods?

File: Presentation.Web/app/components/data-processing/tabs/data-processing-registration-tab-oversight.controller.ts [hunk 4]
```diff
@@ -364,6 +366,26 @@
...
         }
+        
+        private bindScheduledInspectionDate() {
+            this.scheduledInspectionDate = new Models.ViewModel.Generic.DateSelectionViewModel(
+                this.dataProcessingRegistration.oversightScheduledInspectionDate,
+                (newDate) => this.changeScheduledInspectionDate(newDate));
+        }
+
+        private changeScheduledInspectionDate(scheduledInspection: string) {
+            var formattedDate = Helpers.DateStringFormat.fromDDMMYYYYToYYYYMMDD(scheduledInspection);
```
Comment (mrjsawdk) [inline Presentation.Web/app/components/data-processing/tabs/data-processing-registration-tab-oversight.controller.ts:377]: this is probably the cause of it not supporting "reset"

File: Presentation.Web/app/components/data-processing/data-processing-registration-overview.controller.ts [hunk 5]
```diff
@@ -422,6 +422,15 @@
...
                             .withInitialVisibility(false)
                             .withRendering(dataItem => Helpers.RenderFieldsHelper.renderDate(dataItem.LastChangedAt)))
+                    .withColumn(builder =>
+                        builder
+                        .withDataSourceName("OversightScheduledInspectionDate")
+                        .withTitle("Scheduled inspection date title")
+                        .withId("scheduledInspectionDate")
+                        .withDataSourceType(Utility.KendoGrid.KendoGridColumnDataSourceType.Date)
+                        .withFilteringOperation(Utility.KendoGrid.KendoGridColumnFiltering.Date)
+                        .withStandardWidth(170)
```
Comment (mrjsawdk) [inline Presentation.Web/app/components/data-processing/data-processing-registration-overview.controller.ts:432]: make sure the size of the column makes the name fit on one line (use the "reset columns" to see the effects)

File: Presentation.Web/Controllers/API/V1/DataProcessingRegistrationController.cs [hunk 6]
```diff
@@ -662,6 +662,22 @@ public HttpResponseMessage PatchOversightCompleted(int id, [FromBody] SingleValu
...
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.BadRequest)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage PatchOversightScheduledInspectionDate(int id, [FromBody] SingleValueDTO<DateTime?> scheduledInspectionDate)
+        {
+            if (scheduledInspectionDate == null)
+                return BadRequest(nameof(scheduledInspectionDate) + " must be provided");
+
+            return _dataProcessingRegistrationApplicationService
+                .UpdateOversightScheduledInspectionDate(id, scheduledInspectionDate.Value)
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V1/DataProcessingRegistrationController.cs:677]: Simplify it by

```
  return _dataProcessingRegistrationApplicationService
                .UpdateOversightScheduledInspectionDate(id, scheduledInspectionDate.Value)
                .Select(ToDTO)
                .Match(Ok, FromOperationError);
```

**Review by mrjsawdk — APPROVED** [review]

**Pull request merged**